src/Microsoft.Restier.Publishers.OData/RestierController.cs

```diff
@@ -76,16 +80,59 @@ private ApiBase Api
                 throw new InvalidOperationException(Resources.ControllerRequiresPath);
             }
 
+            var result = default(IQueryable);
+
+            // Get queryable path builder to builder
             IQueryable queryable = this.GetQuery();
-            QueryRequest queryRequest = new QueryRequest(queryable)
+
+            // If it is a unbound function, call Api function directly
+            ODataPathSegment lastSegment = path.Segments.LastOrDefault();
+            if (lastSegment != null && lastSegment.SegmentKind != ODataSegmentKinds.UnboundFunction)
```

Comment:
ODataSegmentKinds cannot be used as case condition, and error is "A constant value is expected", I continue use if but extract logic into different method.